pkg/crc/cache/cache.go

```diff
@@ -106,3 +110,20 @@ func (c *Cache) getBinary(destDir string) (string, error) {
 
 	return destPath, err
 }
+
+func (c *Cache) CheckVersion() error {
+	// Check if version string is non-empty
+	if c.version == "" {
+		return nil
+	}
+	// Check the version of a binary matches to supported one
+	binaryPath := filepath.Join(c.destDir, c.binaryName)
+	stdOut, stdErr, err := crcos.RunWithDefaultLocale(binaryPath, c.versionArg...)
```

Comment:
My feeling is that it would be more flexible to pass a `GetVersion` method rather than using `c.VersionArg`. This would give us the opportunity to do more sophisticated version extraction if needed, instead of just `strings.Contains`

Reply:
@cfergeau so `GetVersion` just provide the curent version of binary or will it do something more? `strings.Contains` is part of `CheckVersion` where we compare what is installed vs what we expect.

Reply:
It would return the current version of the binary (which could come from running the binary), then `CheckVersion` could use a semantic version check instead, or use `==` directly, rather than a very wide `strings.Contains`.

Reply:
@cfergeau the only way semantic version check works if each binary we use follows it, currently even the tray bits we control for windows not able to follow semantic versioning.

Reply:
I think the tray versioning is valid semantic versioning, except that the semantic spec explicitly says these are alpha/test releases and don't really define how to compare them ;) Comparing with `==` is still better than `strings.Contains` in my opinion.